app/_layout.tsx

```diff
@@ -3,16 +3,29 @@ import { useFonts } from 'expo-font';
 import { Slot, SplashScreen } from 'expo-router';
 import React, { useEffect, useLayoutEffect, useState } from 'react';
 import type { ColorSchemeName } from 'react-native';
-import { Appearance } from 'react-native';
+import { Appearance, LogBox } from 'react-native';
 import { theme_colors } from '../config/theme';
 import { customFontsToLoad } from '../config/typography';
 import { AuthProvider } from '../context/auth';
 
+LogBox.ignoreLogs([
+  /**
+   * This warning keeps showing up because of a bug in expo.
+   * The PR fixing the bug has been merged but not released yet.
+   * This line removes the warning from the app but it will still show up in the console.
```

Comment:
I had this issue too with expo-router v2. I found a workaround when working with protected routes by checking navigationState. However I don't think my fix would work here so this will do for now. The documentation suggests we use `useFocusEffect` and `router.replace` but that too doesn't solve the warning